solutions/정사각형.js

```diff
@@ -0,0 +1,17 @@
+const INPUT_PATH = "../inputs/정사각형.txt";
+
+const fs = require("fs");
+const filePath = process.platform === "linux" ? "/dev/stdin" : INPUT_PATH;
+const input = fs.readFileSync(filePath).toString().trim().split("\n");
+const [n, m] = input.splice(0, 1)[0].split(" ");
+
+function solution(input) {
+  let result = 1;
+
+  for (let i = 0; i < n; i++) {
+    for (let j = 0; j < m; j++) {
+    }
+  }
+}
```

Comment:
저도 이렇게 접근해서 내부에 for문 한 번 더 썼는데, 옳은 방법인지는 모르겠습니다. 종우님 코드 기준으로 보면,

```suggestion
function solution(input) {
  const square = input.slice(1).map((line) => line.split("").map(Number));
  let result = 1;

  for (let i = 0; i < n; i++) {
    for (let j = 0; j < m; j++) {
      for (let size = 1; i + size < N && j + size < M; size++) {
        if (
          square[i][j] === square[i][j + size] &&
          square[i][j] === square[i + size][j] &&
          square[i][j] === square[i + size][j + size]
        ) {
          const currentSize = (size + 1) * (size + 1);

          if (currentSize > result) {
            result = currentSize;
          }
        }
      }
    }
  }
}
```

이렇게 풀었습니다.
각 i, j 는 왼쪽 위 꼭지점으로 보고 나머지 세개의 꼭지점이 같은지 판단한 후에 크기를 구해서 반환했습니다!